src/Gui/MenuManager.cpp

```diff
@@ -338,6 +340,31 @@ void MenuManager::setup(MenuItem* item, QMenu* menu) const
     }
 }
 
+void MenuManager::setupMenuBarCornerWidgets() const {
+    QMenuBar* menuBar = getMainWindow()->menuBar();
+    ParameterGrp::handle hGrp = WindowParameter::getDefaultParameter()->GetGroup("General");
+
+    //Right corner widget
+    if (hGrp->GetBool("WBSRightCorner", false)) { //add workbench selector to menubar right corner widget.
+        if (!qobject_cast<QComboBox*>(menuBar->cornerWidget(Qt::TopRightCorner)))
+            Application::Instance->commandManager().addTo("Std_Workbench", menuBar);
+    }
+    else if (menuBar->cornerWidget(Qt::TopRightCorner)) {
+        delete menuBar->cornerWidget(Qt::TopRightCorner);
+        menuBar->setCornerWidget(nullptr, Qt::TopRightCorner);

```

Comment:
Why are you deleting ? If someone else use this corner widget for something else ?
Notice correct way to delete widget in Qt is `menuBar->cornerWidget(Qt::TopRightCorner)->deleteLater()`

Reply:
I delete because if someone change the setting from 'right Corner' to toolbar, then the corner widget needs to be removed. Though maybe the removal should be done in the DlgGeneralImp.cpp ?

Reply:
Deleting shall be in the parameter observer.
In order to not delete unexpectedly another widget, you should at least set an objectName to your widget and check it before deletion.